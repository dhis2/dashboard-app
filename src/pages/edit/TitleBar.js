import { OfflineTooltip } from '@dhis2/analytics'
import { useDhis2ConnectionStatus } from '@dhis2/app-runtime'
import i18n from '@dhis2/d2-i18n'
import { InputField, TextAreaField, Radio, Button } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
    acSetDashboardTitle,
    acSetDashboardDescription,
    acSetLayoutColumns,
    tSetDashboardItems,
    acSetItemConfigInsertPosition,
    acSetDashboardCode,
} from '../../actions/editDashboard.js'
import { orObject } from '../../modules/util.js'
import {
    sGetEditDashboardRoot,
    sGetItemConfigInsertPosition,
    sGetLayoutColumns,
} from '../../reducers/editDashboard.js'
import { LayoutFixedIcon } from './assets/LayoutFixed.js'
import { LayoutFreeflowIcon } from './assets/LayoutFreeflow.js'
import ItemSelector from './ItemSelector/ItemSelector.js'
import { LayoutModal } from './LayoutModal.js'
import classes from './styles/TitleBar.module.css'

const EditTitleBar = ({
    insertPosition,
    columns,
    description,
    name,
    code,
    onChangeCode,
    onChangeInsertPosition,
    onChangeTitle,
    onChangeDescription,
    onSaveLayout,
}) => {
    const { isDisconnected: offline } = useDhis2ConnectionStatus()

    const updateTitle = (_, e) => {
        onChangeTitle(e.target.value)
    }

    const updateCode = (_, e) => {
        onChangeCode(e.target.value)
    }

    const updateDescription = (_, e) => {
        onChangeDescription(e.target.value)
    }

    const [showLayoutModal, setShowLayoutModal] = useState(false)

    return (
        <div className={classes.container}>
            <div className={classes.inputWrapper}>
                <div className={classes.inputFieldWrapper}>
                    <InputField
                        name="Dashboard title input"
                        label={i18n.t('Dashboard title')}
                        type="text"
                        onChange={updateTitle}
                        value={name}
                        placeholder={i18n.t('Untitled dashboard')}
                        dataTest="dashboard-title-input"
                        dense
                    />
                    <InputField
                        name="Dashboard code input"
                        label={i18n.t('Dashboard code')}
                        type="text"
                        onChange={updateCode}
                        value={code}
                        dataTest="dashboard-code-input"
                        dense
                        {...(code.length > 50 && {
                            error: true,
                            validationText: i18n.t(
                                "Code can't be longer than 50 characters"
                            ),
                        })}
                    />
                </div>
                <TextAreaField
                    name="Dashboard description input"
                    label={i18n.t('Dashboard description')}
                    onChange={updateDescription}
                    value={description}
                    dataTest="dashboard-description-input"
                    rows={6}
                    dense
                />
            </div>
            <div className={classes.searchContainer}>
                <div className={classes.layoutWrapper}>
                    <p className={classes.label}>{i18n.t('Layout')}</p>
                    <div className={classes.layoutOption}>
                        {columns.length ? (
                            <LayoutFixedIcon />
                        ) : (
                            <LayoutFreeflowIcon />
                        )}
                        <span className={classes.layoutValue}>
                            {columns.length
                                ? i18n.t('{{count}} columns', {
                                      count: columns.length,
                                      defaultValue: '{{count}} column',
                                      defaultValue_plural: '{{count}} columns',
                                  })
                                : i18n.t('Freeflow')}
                        </span>
                        <OfflineTooltip>
                            <Button
                                disabled={offline}
                                small
                                onClick={() => setShowLayoutModal(true)}
                            >
                                {i18n.t('Change layout')}
                            </Button>
                        </OfflineTooltip>
                    </div>
                </div>
                <div className={classes.positionWrapper}>
                    <p className={classes.label}>
                        {i18n.t('Add new items to')}
                    </p>
                    <div className={classes.positionOptions}>
                        <Radio
                            dense
                            label={i18n.t('End of dashboard')}
                            name="END"
                            onChange={(value) =>
                                value.checked &&
                                onChangeInsertPosition(value.name)
                            }
                            checked={insertPosition === 'END'}
                        />
                        <Radio
                            dense
                            label={i18n.t('Start of dashboard')}
                            name="START"
                            onChange={(value) =>
                                value.checked &&
                                onChangeInsertPosition(value.name)
                            }
                            checked={insertPosition === 'START'}
                        />
                    </div>
                </div>
                <div className={classes.searchWrapper}>
                    <ItemSelector />
                </div>
            </div>
            {showLayoutModal && (
                <LayoutModal
                    onClose={() => setShowLayoutModal(false)}
                    columns={columns.length}
                    onSaveLayout={onSaveLayout}
                />
            )}
        </div>
    )
}

EditTitleBar.propTypes = {
    onChangeCode: PropTypes.func.isRequired,
    onChangeDescription: PropTypes.func.isRequired,
    onChangeInsertPosition: PropTypes.func.isRequired,
    onChangeTitle: PropTypes.func.isRequired,
    onSaveLayout: PropTypes.func.isRequired,
    code: PropTypes.string,
    columns: PropTypes.array,
    description: PropTypes.string,
    insertPosition: PropTypes.string,
    name: PropTypes.string,
}

EditTitleBar.defaultProps = {
    name: '',
    code: '',
    description: '',
    insertPosition: 'END',
}

const mapStateToProps = (state) => {
    const selectedDashboard = orObject(sGetEditDashboardRoot(state))

    return {
        name: selectedDashboard.name,
        code: selectedDashboard.code,
        columns: sGetLayoutColumns(state),
        description: selectedDashboard.description,
        insertPosition: sGetItemConfigInsertPosition(state),
    }
}

const mapDispatchToProps = {
    onChangeCode: acSetDashboardCode,
    onChangeTitle: acSetDashboardTitle,
    onChangeDescription: acSetDashboardDescription,
    onChangeInsertPosition: acSetItemConfigInsertPosition,
    onSaveLayout: (columns) => (dispatch) => {
        dispatch(
            acSetLayoutColumns(
                [...Array(columns).keys()].map((i) => ({ index: i }))
            )
        )
        dispatch(tSetDashboardItems())
    },
}

export default connect(mapStateToProps, mapDispatchToProps)(EditTitleBar)
