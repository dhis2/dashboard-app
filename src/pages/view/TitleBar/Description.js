import i18n from '@dhis2/d2-i18n'
import cx from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import classes from './styles/Description.module.css'

const Description = ({ description }) => (
    <div
        className={cx(
            classes.descContainer,
            description ? classes.desc : classes.noDesc
        )}
        data-test="dashboard-description"
    >
        {description || i18n.t('No description')}
    </div>
)

Description.propTypes = {
    description: PropTypes.string,
}

export default Description
