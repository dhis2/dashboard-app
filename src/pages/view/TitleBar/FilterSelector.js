import { DimensionsPanel } from '@dhis2/analytics'
import i18n from '@dhis2/d2-i18n'
import { Card, colors, IconFilter24 } from '@dhis2/ui'
import isEmpty from 'lodash/isEmpty'
import PropTypes from 'prop-types'
import React, { useState } from 'react'
import { connect } from 'react-redux'
import {
    acClearActiveModalDimension,
    acSetActiveModalDimension,
} from '../../../actions/activeModalDimension'
import DropdownButton from '../../../components/DropdownButton/DropdownButton'
import useDimensions from '../../../modules/useDimensions'
import { sGetActiveModalDimension } from '../../../reducers/activeModalDimension'
import { sGetItemFiltersRoot } from '../../../reducers/itemFilters'
import FilterDialog from './FilterDialog'
import classes from './styles/FilterSelector.module.css'

const FilterSelector = props => {
    const [filterDialogIsOpen, setFilterDialogIsOpen] = useState(false)
    const dimensions = useDimensions(filterDialogIsOpen)

    const toggleFilterDialogIsOpen = () =>
        setFilterDialogIsOpen(!filterDialogIsOpen)

    const onCloseDialog = () => {
        setFilterDialogIsOpen(false)

        props.clearActiveModalDimension()
    }

    const selectDimension = id => {
        props.setActiveModalDimension(
            dimensions.find(dimension => dimension.id === id)
        )
    }

    const filterDimensions = () => {
        if (!props.restrictFilters) {
            return dimensions
        } else {
            return dimensions.filter(d =>
                [...props.allowedFilters].includes(d.id)
            )
        }
    }

    const getFilterSelector = () => (
        <Card dataTest="dashboard-filter-popover">
            <DimensionsPanel
                style={{ width: '320px' }}
                dimensions={filterDimensions()}
                onDimensionClick={selectDimension}
                selectedIds={Object.keys(props.initiallySelectedItems)}
            />
        </Card>
    )

    return props.restrictFilters && !props.allowedFilters?.length ? null : (
        <>
            <span className={classes.buttonContainer}>
                <DropdownButton
                    open={filterDialogIsOpen}
                    onClick={toggleFilterDialogIsOpen}
                    icon={<IconFilter24 color={colors.grey700} />}
                    component={getFilterSelector()}
                >
                    {i18n.t('Add filter')}
                </DropdownButton>
            </span>
            {!isEmpty(props.dimension) ? (
                <FilterDialog
                    dimension={props.dimension}
                    onClose={onCloseDialog}
                />
            ) : null}
        </>
    )
}

const mapStateToProps = state => ({
    dimension: sGetActiveModalDimension(state),
    initiallySelectedItems: sGetItemFiltersRoot(state),
})

FilterSelector.propTypes = {
    allowedFilters: PropTypes.array,
    clearActiveModalDimension: PropTypes.func,
    dimension: PropTypes.object,
    initiallySelectedItems: PropTypes.object,
    restrictFilters: PropTypes.bool,
    setActiveModalDimension: PropTypes.func,
}

export default connect(mapStateToProps, {
    clearActiveModalDimension: acClearActiveModalDimension,
    setActiveModalDimension: acSetActiveModalDimension,
})(FilterSelector)
