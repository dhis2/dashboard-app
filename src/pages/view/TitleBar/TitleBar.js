import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import {
    sGetSelectedDisplayName,
    sGetSelectedDisplayDescription,
} from '../../../reducers/selected'
import { sGetShowDescription } from '../../../reducers/showDescription'
import ActionsBar from './ActionsBar'
import Description from './Description'
import classes from './styles/TitleBar.module.css'

const ViewTitleBar = ({ name, description, showDescription }) => (
    <div className={classes.container}>
        <div className={classes.titleBar} data-test="title-bar">
            <span className={classes.title} data-test="view-dashboard-title">
                {name}
            </span>
            <ActionsBar />
        </div>
        {showDescription && <Description description={description} />}
    </div>
)

ViewTitleBar.propTypes = {
    description: PropTypes.string,
    name: PropTypes.string,
    showDescription: PropTypes.bool,
}

const mapStateToProps = state => ({
    name: sGetSelectedDisplayName(state),
    description: sGetSelectedDisplayDescription(state),
    showDescription: sGetShowDescription(state),
})

export default connect(mapStateToProps)(ViewTitleBar)
