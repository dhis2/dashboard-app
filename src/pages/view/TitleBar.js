import React, { useState } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { Link, Redirect } from 'react-router-dom'
import i18n from '@dhis2/d2-i18n'
import SharingDialog from '@dhis2/d2-ui-sharing-dialog'
import { useDataEngine, useAlert } from '@dhis2/app-runtime'
import {
    Button,
    FlyoutMenu,
    MenuItem,
    Tooltip,
    colors,
    IconMore24,
    IconStar24,
    IconStarFilled24,
} from '@dhis2/ui'
import { useD2 } from '@dhis2/app-runtime-adapter-d2'

import { orObject } from '../../modules/util'
import { apiStarDashboard } from './starDashboard'
import { apiPostShowDescription } from '../../api/description'

import { acSetDashboardStarred } from '../../actions/dashboards'
import { acSetShowDescription } from '../../actions/showDescription'
import FilterSelector from './ItemFilter/FilterSelector'
import DropdownButton from '../../components/DropdownButton/DropdownButton'
import { sGetSelectedId } from '../../reducers/selected'
import { sGetShowDescription } from '../../reducers/showDescription'
import {
    sGetDashboardById,
    sGetDashboardItems,
    EMPTY_DASHBOARD,
} from '../../reducers/dashboards'

import classes from './styles/TitleBar.module.css'

const ViewTitleBar = ({
    id,
    name,
    description,
    access,
    showDescription,
    starred,
    setDashboardStarred,
    updateShowDescription,
    restrictFilters,
    allowedFilters,
}) => {
    const [moreOptionsSmallIsOpen, setMoreOptionsSmallIsOpen] = useState(false)
    const [moreOptionsIsOpen, setMoreOptionsIsOpen] = useState(false)
    const [sharingDialogIsOpen, setSharingDialogIsOpen] = useState(false)
    const [redirectUrl, setRedirectUrl] = useState(null)
    const { d2 } = useD2()
    const dataEngine = useDataEngine()

    const warningAlert = useAlert(({ msg }) => msg, {
        warning: true,
    })

    const toggleSharingDialog = () =>
        setSharingDialogIsOpen(!sharingDialogIsOpen)

    const toggleMoreOptions = small =>
        small
            ? setMoreOptionsSmallIsOpen(!moreOptionsSmallIsOpen)
            : setMoreOptionsIsOpen(!moreOptionsIsOpen)

    const printLayout = () => setRedirectUrl(`${id}/printlayout`)
    const printOipp = () => setRedirectUrl(`${id}/printoipp`)

    const StarIcon = starred ? IconStarFilled24 : IconStar24

    if (redirectUrl) {
        return <Redirect to={redirectUrl} />
    }

    const showHideDescriptionLabel = showDescription
        ? i18n.t('Hide description')
        : i18n.t('Show description')

    const onToggleShowDescription = () =>
        apiPostShowDescription(!showDescription)
            .then(() => {
                updateShowDescription(!showDescription)
                toggleMoreOptions()
            })
            .catch(() => {
                const msg = showDescription
                    ? i18n.t('Failed to hide description')
                    : i18n.t('Failed to show description')
                warningAlert.show({ msg })
            })

    const toggleStarredDashboardLabel = starred
        ? i18n.t('Unstar dashboard')
        : i18n.t('Star dashboard')

    const onToggleStarredDashboard = () =>
        apiStarDashboard(dataEngine, id, !starred)
            .then(() => {
                setDashboardStarred(id, !starred)
                if (moreOptionsIsOpen) {
                    toggleMoreOptions()
                }
            })
            .catch(() => {
                const msg = starred
                    ? i18n.t('Failed to unstar the dashboard')
                    : i18n.t('Failed to star the dashboard')
                warningAlert.show({ msg })
            })

    const userAccess = orObject(access)

    const descriptionClasses = cx(
        classes.descContainer,
        description ? classes.desc : classes.noDesc
    )

    const getMoreMenu = () => (
        <FlyoutMenu>
            <MenuItem
                dense
                label={toggleStarredDashboardLabel}
                onClick={onToggleStarredDashboard}
            />
            <MenuItem
                dense
                label={showHideDescriptionLabel}
                onClick={onToggleShowDescription}
            />
            <MenuItem dense label={i18n.t('Print')} dataTest="print-menu-item">
                <MenuItem
                    dense
                    label={i18n.t('Dashboard layout')}
                    onClick={printLayout}
                    dataTest="print-layout-menu-item"
                />
                <MenuItem
                    dense
                    label={i18n.t('One item per page')}
                    onClick={printOipp}
                    dataTest="print-oipp-menu-item"
                />
            </MenuItem>
        </FlyoutMenu>
    )

    const getMoreButton = (className, useSmall) => (
        <DropdownButton
            className={className}
            small={useSmall}
            onClick={() => toggleMoreOptions(useSmall)}
            icon={<IconMore24 color={colors.grey700} />}
            component={getMoreMenu()}
            open={useSmall ? moreOptionsSmallIsOpen : moreOptionsIsOpen}
        >
            {i18n.t('More')}
        </DropdownButton>
    )

    return (
        <>
            <div className={classes.container}>
                <div className={classes.titleBar}>
                    <span
                        className={classes.title}
                        data-test="view-dashboard-title"
                    >
                        {name}
                    </span>
                    <div className={classes.actions}>
                        <div
                            className={classes.star}
                            onClick={onToggleStarredDashboard}
                            data-test="button-star-dashboard"
                        >
                            <Tooltip
                                content={
                                    starred
                                        ? i18n.t('Unstar dashboard')
                                        : i18n.t('Star dashboard')
                                }
                            >
                                <span
                                    data-test={
                                        starred
                                            ? 'dashboard-starred'
                                            : 'dashboard-unstarred'
                                    }
                                >
                                    <StarIcon color={colors.grey600} />
                                </span>
                            </Tooltip>
                        </div>
                        <div className={classes.strip}>
                            {userAccess.update ? (
                                <Link
                                    className={classes.editLink}
                                    to={`/${id}/edit`}
                                    data-test="link-edit-dashboard"
                                >
                                    <Button>{i18n.t('Edit')}</Button>
                                </Link>
                            ) : null}
                            {userAccess.manage ? (
                                <Button
                                    className={classes.shareButton}
                                    onClick={toggleSharingDialog}
                                >
                                    {i18n.t('Share')}
                                </Button>
                            ) : null}
                            <FilterSelector
                                allowedFilters={allowedFilters}
                                restrictFilters={restrictFilters}
                            />
                            {getMoreButton(classes.moreButton, false)}
                            {getMoreButton(classes.moreButtonSmall, true)}
                        </div>
                    </div>
                </div>
                {showDescription && (
                    <div
                        className={descriptionClasses}
                        data-test="dashboard-description"
                    >
                        {description || i18n.t('No description')}
                    </div>
                )}
            </div>
            {id && (
                <SharingDialog
                    d2={d2}
                    id={id}
                    type="dashboard"
                    open={sharingDialogIsOpen}
                    onRequestClose={toggleSharingDialog}
                    insertTheme={true}
                />
            )}
        </>
    )
}

ViewTitleBar.propTypes = {
    access: PropTypes.object,
    allowedFilters: PropTypes.array,
    description: PropTypes.string,
    id: PropTypes.string,
    name: PropTypes.string,
    restrictFilters: PropTypes.bool,
    setDashboardStarred: PropTypes.func,
    showDescription: PropTypes.bool,
    starred: PropTypes.bool,
    updateShowDescription: PropTypes.func,
}

ViewTitleBar.defaultProps = {
    name: '',
    description: '',
    starred: false,
    showDescription: false,
}

const mapStateToProps = state => {
    const id = sGetSelectedId(state)
    const dashboard = sGetDashboardById(state, id) || EMPTY_DASHBOARD

    return {
        id,
        name: dashboard.displayName,
        description: dashboard.displayDescription,
        dashboardItems: sGetDashboardItems(state),
        showDescription: sGetShowDescription(state),
        starred: dashboard.starred,
        access: dashboard.access,
        restrictFilters: dashboard.restrictFilters,
        allowedFilters: dashboard.allowedFilters,
    }
}

export default connect(mapStateToProps, {
    setDashboardStarred: acSetDashboardStarred,
    updateShowDescription: acSetShowDescription,
})(ViewTitleBar)
