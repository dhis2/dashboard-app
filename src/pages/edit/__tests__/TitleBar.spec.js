import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import TitleBar from '../TitleBar'

const mockStore = configureMockStore()

jest.mock(
    '../ItemSelector/ItemSelector',
    () =>
        function MockItemSelector() {
            return <div className="item-selector" />
        }
)

describe('TitleBar', () => {
    it('renders correctly with name and description', () => {
        const store = {
            editDashboard: {
                name: 'Rainbow Dash',
                description: 'A very colorful pony',
            },
        }
        const { container } = render(
            <Provider store={mockStore(store)}>
                <TitleBar />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })

    it('renders correctly when no name or description', () => {
        const store = {
            editDashboard: {
                name: '',
                description: '',
            },
        }

        const { container } = render(
            <Provider store={mockStore(store)}>
                <TitleBar />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
