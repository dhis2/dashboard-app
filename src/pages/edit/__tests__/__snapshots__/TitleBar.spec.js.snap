// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar renders correctly when no name or description 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="inputWrapper"
    >
      <div
        class="jsx-1141087211 "
        data-test="dashboard-title-input"
      >
        <label
          class="jsx-2718078600 "
          data-test="dashboard-title-input-label"
          for="Dashboard title input"
        >
          <span
            class="jsx-2718078600"
          >
            Dashboard title
          </span>
        </label>
        <div
          class="jsx-3111771310 "
          data-test="dashboard-title-input-content"
        >
          <div
            class="jsx-1220078726 "
            data-test="dhis2-uicore-box"
          >
            <div
              class="jsx-3353877153 jsx-2133963083 input"
              data-test="dhis2-uicore-input"
            >
              <input
                class="jsx-3353877153 jsx-2133963083 "
                id="Dashboard title input"
                name="Dashboard title input"
                placeholder="Untitled dashboard"
                type="text"
                value=""
              />
              <div
                class="jsx-3353877153 jsx-2133963083 status-icon"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="jsx-1141087211 "
        data-test="dashboard-description-input"
      >
        <label
          class="jsx-2718078600 "
          data-test="dashboard-description-input-label"
          for="Dashboard description input"
        >
          <span
            class="jsx-2718078600"
          >
            Dashboard description
          </span>
        </label>
        <div
          class="jsx-3111771310 "
          data-test="dashboard-description-input-content"
        >
          <div
            class="jsx-2954617683 "
            data-test="dhis2-uicore-box"
          >
            <div
              class="jsx-4150870461 jsx-2961587785 textarea"
              data-test="dhis2-uicore-textarea"
            >
              <textarea
                class="jsx-4150870461 jsx-2961587785 "
                id="Dashboard description input"
                name="Dashboard description input"
                rows="4"
              />
              <div
                class="jsx-4150870461 jsx-2961587785 status-icon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="searchContainer"
    >
      <div
        class="layoutWrapper"
      >
        <p
          class="label"
        >
          Layout
        </p>
        <div
          class="layoutOption"
        >
          <svg
            height="42"
            viewBox="0 0 42 42"
            width="42"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m42 33v9h-42v-9zm-31-12v7h-11v-7zm15 0v7h-11v-7zm16-21v28h-10v-28zm-16 0v16h-26v-16z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="layoutValue"
          >
            Freeflow
          </span>
          <button
            class="jsx-1937199209 small"
            data-test="dhis2-uicore-button"
            type="button"
          >
            Change layout
          </button>
        </div>
      </div>
      <div
        class="positionWrapper"
      >
        <p
          class="label"
        >
          Add new items to
        </p>
        <div
          class="positionOptions"
        >
          <label
            class="jsx-869913229 dense"
            data-test="dhis2-uicore-radio"
          >
            <input
              checked=""
              class="jsx-869913229"
              name="END"
              type="radio"
              value=""
            />
            <div
              class="jsx-869913229 icon dense"
            >
              <svg
                class="jsx-1659897952 jsx-1631270792 checked"
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="jsx-1659897952 jsx-1631270792 background"
                  cx="7"
                  cy="7"
                  r="7"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer unchecked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,13 C3.6862915,13 1,10.3137085 1,7 C1,3.6862915 3.6862915,1 7,1 C10.3137085,1 13,3.6862915 13,7 C13,10.3137085 10.3137085,13 7,13 Z"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer checked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,12 C4.23857625,12 2,9.76142375 2,7 C2,4.23857625 4.23857625,2 7,2 C9.76142375,2 12,4.23857625 12,7 C12,9.76142375 9.76142375,12 7,12 Z"
                />
                <circle
                  class="jsx-1659897952 jsx-1631270792 inner"
                  cx="7"
                  cy="7"
                  r="3"
                />
              </svg>
            </div>
            End of dashboard
          </label>
          <label
            class="jsx-869913229 dense"
            data-test="dhis2-uicore-radio"
          >
            <input
              class="jsx-869913229"
              name="START"
              type="radio"
              value=""
            />
            <div
              class="jsx-869913229 icon dense"
            >
              <svg
                class="jsx-1659897952 jsx-1631270792 "
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="jsx-1659897952 jsx-1631270792 background"
                  cx="7"
                  cy="7"
                  r="7"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer unchecked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,13 C3.6862915,13 1,10.3137085 1,7 C1,3.6862915 3.6862915,1 7,1 C10.3137085,1 13,3.6862915 13,7 C13,10.3137085 10.3137085,13 7,13 Z"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer checked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,12 C4.23857625,12 2,9.76142375 2,7 C2,4.23857625 4.23857625,2 7,2 C9.76142375,2 12,4.23857625 12,7 C12,9.76142375 9.76142375,12 7,12 Z"
                />
                <circle
                  class="jsx-1659897952 jsx-1631270792 inner"
                  cx="7"
                  cy="7"
                  r="3"
                />
              </svg>
            </div>
            Start of dashboard
          </label>
        </div>
      </div>
      <div
        class="searchWrapper"
      >
        <div
          class="item-selector"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TitleBar renders correctly with name and description 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="inputWrapper"
    >
      <div
        class="jsx-1141087211 "
        data-test="dashboard-title-input"
      >
        <label
          class="jsx-2718078600 "
          data-test="dashboard-title-input-label"
          for="Dashboard title input"
        >
          <span
            class="jsx-2718078600"
          >
            Dashboard title
          </span>
        </label>
        <div
          class="jsx-3111771310 "
          data-test="dashboard-title-input-content"
        >
          <div
            class="jsx-1220078726 "
            data-test="dhis2-uicore-box"
          >
            <div
              class="jsx-3353877153 jsx-2133963083 input"
              data-test="dhis2-uicore-input"
            >
              <input
                class="jsx-3353877153 jsx-2133963083 "
                id="Dashboard title input"
                name="Dashboard title input"
                placeholder="Untitled dashboard"
                type="text"
                value="Rainbow Dash"
              />
              <div
                class="jsx-3353877153 jsx-2133963083 status-icon"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="jsx-1141087211 "
        data-test="dashboard-description-input"
      >
        <label
          class="jsx-2718078600 "
          data-test="dashboard-description-input-label"
          for="Dashboard description input"
        >
          <span
            class="jsx-2718078600"
          >
            Dashboard description
          </span>
        </label>
        <div
          class="jsx-3111771310 "
          data-test="dashboard-description-input-content"
        >
          <div
            class="jsx-2954617683 "
            data-test="dhis2-uicore-box"
          >
            <div
              class="jsx-4150870461 jsx-2961587785 textarea"
              data-test="dhis2-uicore-textarea"
            >
              <textarea
                class="jsx-4150870461 jsx-2961587785 "
                id="Dashboard description input"
                name="Dashboard description input"
                rows="4"
              >
                A very colorful pony
              </textarea>
              <div
                class="jsx-4150870461 jsx-2961587785 status-icon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="searchContainer"
    >
      <div
        class="layoutWrapper"
      >
        <p
          class="label"
        >
          Layout
        </p>
        <div
          class="layoutOption"
        >
          <svg
            height="42"
            viewBox="0 0 42 42"
            width="42"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m42 33v9h-42v-9zm-31-12v7h-11v-7zm15 0v7h-11v-7zm16-21v28h-10v-28zm-16 0v16h-26v-16z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="layoutValue"
          >
            Freeflow
          </span>
          <button
            class="jsx-1937199209 small"
            data-test="dhis2-uicore-button"
            type="button"
          >
            Change layout
          </button>
        </div>
      </div>
      <div
        class="positionWrapper"
      >
        <p
          class="label"
        >
          Add new items to
        </p>
        <div
          class="positionOptions"
        >
          <label
            class="jsx-869913229 dense"
            data-test="dhis2-uicore-radio"
          >
            <input
              checked=""
              class="jsx-869913229"
              name="END"
              type="radio"
              value=""
            />
            <div
              class="jsx-869913229 icon dense"
            >
              <svg
                class="jsx-1659897952 jsx-1631270792 checked"
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="jsx-1659897952 jsx-1631270792 background"
                  cx="7"
                  cy="7"
                  r="7"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer unchecked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,13 C3.6862915,13 1,10.3137085 1,7 C1,3.6862915 3.6862915,1 7,1 C10.3137085,1 13,3.6862915 13,7 C13,10.3137085 10.3137085,13 7,13 Z"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer checked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,12 C4.23857625,12 2,9.76142375 2,7 C2,4.23857625 4.23857625,2 7,2 C9.76142375,2 12,4.23857625 12,7 C12,9.76142375 9.76142375,12 7,12 Z"
                />
                <circle
                  class="jsx-1659897952 jsx-1631270792 inner"
                  cx="7"
                  cy="7"
                  r="3"
                />
              </svg>
            </div>
            End of dashboard
          </label>
          <label
            class="jsx-869913229 dense"
            data-test="dhis2-uicore-radio"
          >
            <input
              class="jsx-869913229"
              name="START"
              type="radio"
              value=""
            />
            <div
              class="jsx-869913229 icon dense"
            >
              <svg
                class="jsx-1659897952 jsx-1631270792 "
                viewBox="0 0 14 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="jsx-1659897952 jsx-1631270792 background"
                  cx="7"
                  cy="7"
                  r="7"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer unchecked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,13 C3.6862915,13 1,10.3137085 1,7 C1,3.6862915 3.6862915,1 7,1 C10.3137085,1 13,3.6862915 13,7 C13,10.3137085 10.3137085,13 7,13 Z"
                />
                <path
                  class="jsx-1659897952 jsx-1631270792 outer checked"
                  d="M7,14 C10.8659932,14 14,10.8659932 14,7 C14,3.13400675 10.8659932,0 7,0 C3.13400675,0 0,3.13400675 0,7 C0,10.8659932 3.13400675,14 7,14 Z M7,12 C4.23857625,12 2,9.76142375 2,7 C2,4.23857625 4.23857625,2 7,2 C9.76142375,2 12,4.23857625 12,7 C12,9.76142375 9.76142375,12 7,12 Z"
                />
                <circle
                  class="jsx-1659897952 jsx-1631270792 inner"
                  cx="7"
                  cy="7"
                  r="3"
                />
              </svg>
            </div>
            Start of dashboard
          </label>
        </div>
      </div>
      <div
        class="searchWrapper"
      >
        <div
          class="item-selector"
        />
      </div>
    </div>
  </div>
</div>
`;
